sources/org.osbuild.skopeo

```diff
@@ -103,17 +107,21 @@ class SkopeoSource(sources.SourceService):
                            check=True)
 
             # Verify that the digest supplied downloaded the correct container image id.
-            # The image id is the digest of the config, but skopeo can' currently
+            # The image id is the digest of the config, but skopeo can't currently
             # get the config id, only the full config, so we checksum it ourselves.
             res = subprocess.check_output(["skopeo", "inspect", "--raw", "--config", destination])
             downloaded_id = "sha256:" + hashlib.sha256(res).hexdigest()
             if downloaded_id != image_id:
                 raise RuntimeError(
                     f"Downloaded image {imagename}@{digest} has a id of {downloaded_id}, but expected {image_id}")
 
-            # Atomically move download dir into place on successful download
+            # Atomically move download archive into place on successful download
             with ctx.suppress_oserror(errno.ENOTEMPTY, errno.EEXIST):
-                os.rename(archive_dir, f"{self.cache}/{image_id}")
+                os.makedirs(f"{self.cache}/{image_id}", exist_ok=True)
+                os.rename(f"{archive_dir}/{self.dir_name}", f"{self.cache}/{image_id}/{self.dir_name}")
```

Comment:
Please use path concatenation functions on the above two lines instead of string formatting, either `os.path.join` or `pathlib` would be fine.